tests/testthat/test-rename.r

```diff
@@ -37,3 +38,61 @@ test_that("Renaming lists", {
   y <- rename(x, c("c" = "f", "b" = "e", "a" = "d"))
   expect_identical(y, list(d = 1, e = 2, f = 3))
 })
+
+# Duplicate Names -----------------------------------
+context("Rename - Duplicates")
+
+# This batch tests the typical renaming scenarios
+test_that("Renaming list with an conflicting variable name - default", {
+  x <- list(a = 1, b = 2, c = 3)
+  replace_list <- c("c" = "f", "b" = "e", "a" = "f")
+  expected_response <- "The plyr::rename operation has created duplicates for the following name\\(s\\): \\(`f`\\)"
+  expect_warning(object = rename(x = x, replace = replace_list), regexp = expected_response)
```

Comment:
You don't need to name the arguments to expectations, and I'd shrink the match to "created duplicates"